### ISSUE TITLE
java.lang.ClassCastException on Image Blob

### ISSUE THREAD

**Issue by DiegoTavares** [Raizlabs/DBFlow#627]
I'm getting the error: 
 java.lang.ClassCastException: android.graphics.Bitmap cannot be cast to byte[]
when trying to retrieve an element that has a blob element by Foreign Key Container.

Model:

```java
@ModelContainer
@Table(database = CoffeeDatabase.class)
public class User extends BaseModel implements UserInterface{

    @PrimaryKey(autoincrement = true)
    private long id;

    @Column
    @NotNull
    private String name;

    @Column
    @NotNull
    private String email;

    @Column(typeConverter = ImageConverter.class)
    public Bitmap image;
    [...]
}
```

Here is my image converter:

```java
@com.raizlabs.android.dbflow.annotation.TypeConverter
public class ImageConverter extends TypeConverter<Blob, Bitmap> {

    @Override
    public Blob getDBValue(Bitmap model) {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        model.compress(Bitmap.CompressFormat.PNG, 0, baos);

        return new Blob(baos.toByteArray());
    }

    @Override
    public Bitmap getModelValue(Blob data) {
        byte[] blob = data.getBlob();
        return BitmapFactory.decodeByteArray(blob, 0, blob.length);
    }
}
```

The exception is thrown when executing:

```java
        if(user == null && userForeignKeyContainer != null) {
            user = userForeignKeyContainer.toModel();
        }
```

The generated user container "toModel()" function is calling getBlbValue("image") which tries to cast the bitmap value to byte[].

in User_Container.java:
```java
@Override
  public final User toModel(ModelContainer<User, ?> modelContainer) {
    User model = new User();
    model.setId(modelContainer.getLngValue("id"));
    model.setName(modelContainer.getStringValue("name"));
    model.setEmail(modelContainer.getStringValue("email"));
    model.image = typeConverterImageConverter.getModelValue((com.raizlabs.android.dbflow.data.Blob) new Blob(modelContainer.getBlbValue("image")));
    return model;
  }
```

It seems that the converter is not being called.

**Comment by gauravsak** [comment]
Can you post the whole stack trace please?

**Comment by agrosner** [comment]
should be fixed in a 3.0.0-beta4 :)
